src/ResourceManager/Storage/Commands.Management.Storage/StorageAccount/SetAzureStorageAccount.cs

```diff
@@ -123,6 +124,34 @@ public bool EnableHttpsTrafficOnly
 
         private bool? enableHttpsTrafficOnly = null;
 
+        [Parameter(
+            Mandatory = false,
+            ValueFromPipeline = true,
+            HelpMessage = "Storage Account Encryption KeySource.")]
+        public PSKeySource KeySource
```

Comment:
Based on the way I suspect this works (KeySource is either Storage or KeyVault) why would I ever set a Storage KeySource explicitly?

Also, why not create a seperate parameter set that eimply allows setting the 3 KeyVault settings, or allows settign the KeySource to Storage?  Requiring piping seems unnecessary